llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp

```diff
@@ -2514,9 +2514,11 @@ Instruction *InstCombinerImpl::visitCallInst(CallInst &CI) {
     Value* Arg = II->getArgOperand(0);
     Value* X;
     // fabs (-X) --> fabs (X)
-    if (match(Arg, m_FNeg(m_Value(X))))
-        return replaceInstUsesWith(
-            CI, Builder.CreateUnaryIntrinsic(Intrinsic::fabs, X));
+    if (match(Arg, m_FNeg(m_Value(X)))) {
+      CallInst *Fabs = Builder.CreateUnaryIntrinsic(Intrinsic::fabs, X);
+      Fabs->copyFastMathFlags(II);
```

Comment:
You can also pass the II as the 3rd argument to CreateUnaryIntrinsic as the FMFSource (though this will need adjustment depending on how this is extended to handle copysign)

Reply:
Changed the code to pass II as the 3rd argument to CreateUnaryIntrinsic. Made necessary adjustments